### ISSUE TITLE
No such file or directory

### ISSUE THREAD

**Issue by pepealonso95** [tanersener/flutter-ffmpeg#88]
While compressing on ios i keep getting this issue 

```flutter: /data/user/0/myapp/app_flutter/Videos/memory-1574781250937.mp4: No such file or directory```

I checked the directory couples of times and even created the directory previous to compressing and it does exist. Everything works just fine on android.

```

Running FFmpeg with arguments: (
    "-i",
    "/Users/user/Library/Developer/CoreSimulator/Devices/372F619C-CF7F-4CD0-8EF9-ED46C7024EF6/data/Containers/Data/Application/18BDE5AC-E207-4BF8-BC81-C0FB8069F79D/Library/Caches/libCachedImageData/a76249f0-0e3a-11ea-e43b-a3edc3c0d82d.mp4",
    "-c:v",
    libx265,
    "-crf",
    28,
    "-preset",
    ultrafast,
    "/data/user/0/myapp/app_flutter/Videos/memory-1574781250937.mp4"
).
FFmpeg exited with rc: 1

flutter:   Metadata:
flutter:     major_brand     :
flutter: mp42
flutter:
flutter:     minor_version   :
flutter: 0
flutter:
flutter:     compatible_brands:
flutter: mp42mp41isomavc1
flutter:
flutter:     creation_time   :
flutter: 2015-08-10T09:53:36.000000Z
flutter:
flutter:   Duration:
flutter: 00:00:07.54
flutter: , start:
flutter: 0.000000
flutter: , bitrate:
flutter: 2578 kb/s
flutter:
flutter:     Stream #0:0
flutter: (und)
flutter: : Video: h264 (avc1 / 0x31637661), yuv420p(tv, bt709), 1280x720 [SAR 1:1 DAR 16:9], 2424 kb/s
flutter: ,
flutter: 29.97 fps,
flutter: 29.97 tbr,
flutter: 30k tbn,
flutter: 59.94 tbc
flutter:  (default)
flutter:
flutter:     Metadata:
flutter:       creation_time   :
flutter: 2015-08-10T09:53:36.000000Z
flutter:
flutter:       handler_name    :
flutter: L-SMASH Video Handler
flutter:
flutter:       encoder         :
flutter: AVC Coding
flutter:
flutter:     Stream #0:1
flutter: (und)
flutter: : Audio: aac (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 147 kb/s
flutter:  (default)
flutter:
flutter:     Metadata:
flutter:       creation_time   :
flutter: 2015-08-10T09:53:36.000000Z
flutter:
flutter:       handler_name    :
flutter: L-SMASH Audio Handler
flutter:
flutter: /data/user/0/myapp/app_flutter/Videos/memory-1574781250937.mp4: No such file or directory
```

**Comment by tanersener** [comment]
Path format of your output file `/data/user/0/myapp/app_flutter/Videos/memory-1574781250937.mp4` is  Android specific. 

Can you create your output file in one of the folders defined by [path_provider](https://pub.dev/packages/path_provider) plugin?

**Comment by pepealonso95** [comment]
I create the directory using the path provider plugin
    
 ``` final Directory appDirectory = await getApplicationDocumentsDirectory();
      final String videoDirectory = '${appDirectory.path}/Videos';
      Directory dir = await Directory(videoDirectory).create(recursive: true);
      bool ex = dir.existsSync();```

**Comment by tanersener** [comment]
`path_provider` plugin does not define an `appDirectory`. Are you sure you are using `path_provider`?

**Comment by pepealonso95** [comment]
Yes i am, i'm sorry i did not copy the complete example properly..
```
      final Directory appDirectory = await getApplicationDocumentsDirectory();
      final String videoDirectory = '${appDirectory.path}/Videos';
      Directory dir = await Directory(videoDirectory).create(recursive: true);
      bool ex = dir.existsSync();
```

**Comment by tanersener** [comment]
What is the path defined by `videoDirectory`?

**Comment by pepealonso95** [comment]
it ends up being 
```
/data/user/0/myapp/app_flutter/Videos
```

**Comment by tanersener** [comment]
Can you try the following snippet to get application directory?

```
  Future<String> getMyAppDirectory() async {
    final Directory appDirectory = await getApplicationDocumentsDirectory();
    return appDirectory.path;
  }

  this.getMyAppDirectory().then((path) {
    print('My app directory is ' + path);
  });
```

`flutter: My app directory is /Users/taner/Library/Developer/CoreSimulator/Devices/687C6C81-4735-1FAF-3111-8145D028A7E3/data/Containers/Data/Application/DF607511-C337-413B-8002-E2D0A2BC4AB0/Documents`

**Comment by philalem** [comment]
I'm also having a similar issue. flutter-ffmpeg is not recognizing the path to the video I have stored from my `camera` controller. The weird thing is that it plays the .mp4 file via the `video_player` package and it saves properly to google cloud storage, but I get the same `No such file or directory` error when I try to access it via flutter-ffmpeg.

For reference, I used `getApplicationDocumentsDirectory()` to create the directory to my video. My path looks something like this: 
`/var/mobile/Containers/Data/Application/C7EF-4449-FD2D197FACA/Documents/1234/user-posts/1593189772144.mp4`

Also, this is the code relating to flutter-ffmpeg:
```
var arguments = [
      "-y",
      "-i",
      "${widget.paths[0]}",
      "-vframes",
      "1",
      "-an",
      "-s",
      "-ss",
      "1",
      "$thumbnailPath"
    ];
    await _flutterFFmpeg
        .executeWithArguments(arguments)
        .then((rc) => print("FFmpeg process exited with rc $rc"));
```
